### PR TITLE
Pin 1315 authorization-process GET /clients

### PR DESCRIPTION
**Pull request by nttdata-rtorsoli**

### CODE REVIEW

File: src/main/resources/interface-specification.yml [hunk 1]
```diff
@@ -6325,6 +6325,161 @@ paths:
...
+    parameters:
+      - $ref: '#/components/parameters/CorrelationIdHeader'
+      - $ref: '#/components/parameters/IpAddress'
+    get:
+      tags:
+        - clients
+      summary: retrieves a list of clients
+      description: retrieves a list of clients
+      operationId: getClients
+      parameters:
```
Comment (galales) [inline src/main/resources/interface-specification.yml:6338]: let's use only the parameters requested in the story, and hide the others in the implementation

**Review by galales — CHANGES_REQUESTED** [review]

File: src/main/resources/interface-specification.yml [hunk 2]
```diff
@@ -8132,6 +8287,29 @@ components:
...
+        results:
+          type: array
+          items:
+            $ref: '#/components/schemas/CompactClient'
+        pagination:
+          $ref: '#/components/schemas/Pagination'
+      required:
+        - results
+        - pagination    
+    CompactClient:
```
Comment (galales) [inline src/main/resources/interface-specification.yml]: I just noticed that a model `Client` already exists (it just has 1 more field).
We could rename that one to `CompactClient` and use it here

File: src/main/resources/interface-specification.yml [hunk 3]
```diff
@@ -8132,6 +8287,29 @@ components:
           type: boolean
         activeDescriptor:
           $ref: '#/components/schemas/CompactDescriptor'
+    Clients:
```
Comment (galales) [inline src/main/resources/interface-specification.yml]:
```suggestion
    CompactClients:
```

File: src/main/resources/interface-specification.yml (hunk 1 above)
Comment (TonioGela) [reply src/main/resources/interface-specification.yml:6338]: +1

File: src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/ClientsApiServiceImpl.scala [hunk 4]
```diff
@@ -1,5 +1,6 @@
 package it.pagopa.interop.backendforfrontend.api.impl
 
+import cats.implicits._
```
Comment (TonioGela) [inline src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/ClientsApiServiceImpl.scala]:
```suggestion
import cats.syntax.all._
```

`cats.implicits._` is deprecated: https://typelevel.org/cats/imports.html

File: src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/ClientsApiServiceImpl.scala [hunk 5]
```diff
@@ -206,4 +208,40 @@ final case class ClientsApiServiceImpl(authorizationProcessService: Authorizatio
...
+    limit: Int
+  )(implicit
+    contexts: Seq[(String, String)],
+    toEntityMarshallerClients: ToEntityMarshaller[Clients],
+    toEntityMarshallerProblem: ToEntityMarshaller[Problem]
+  ): Route = {
+    val result: Future[Clients] = for {
+      relationshipsUuid <- parseArrayParameters(relationshipIds).traverse(_.toFutureUUID)
+      consumerUuid      <- consumerId.toFutureUUID
+      purposeUuid       <- purposeId.map(_.toFutureUUID).sequence
```
Comment (TonioGela) [inline src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/ClientsApiServiceImpl.scala]:
```suggestion
      purposeUuid       <- purposeId.traverse(_.toFutureUUID)
```

**Review by TonioGela — CHANGES_REQUESTED** [review]

File: src/main/resources/interface-specification.yml (hunk 1 above)
Comment (nttdata-rtorsoli) [reply src/main/resources/interface-specification.yml:6338]: Apologize me, but I have some questions about this issue:
consumerId: is mandatory how can achieve it without pass as a parameter?
purposeId: is not mandatory so I can pass None
kind: the same thing above I can pass None


Can you enlighten me on the situation?

Comment (galales) [reply src/main/resources/interface-specification.yml:6338]: for the `consumerId` you can use the `organizationId` in the jwt.
I spoke with the front, let's leave the `kind` parameter, which is actually useful

File: src/main/resources/interface-specification.yml [hunk 6]
```diff
@@ -6325,6 +6325,148 @@ paths:
...
+            items:
+              type: string
+              format: uuid
+            default: []
+          explode: false    
+        - name: kind
+          in: query
+          description: type of Client to be retrieved
+          schema:
+            type: string
```
Comment (galales) [inline src/main/resources/interface-specification.yml]: `kind` parameter should be an enum

**Review by galales — CHANGES_REQUESTED** [review]

**Review by galales — APPROVED** [review]

**Review by TonioGela — APPROVED** [review]

**Pull request merged**